api/v1alpha1/addresspool_conversion_test.go

```diff
@@ -30,11 +30,11 @@ func TestValidateAddressPoolConvertTo(t *testing.T) {
 			},
 
 			Protocol:   "bgp",
-			AutoAssign: pointer.BoolPtr(false),
+			AutoAssign: ptr.To[bool](false),
```

Comment:
Stumbled in to this PR, and i wondered why you chose to do:
```
ptr.To[bool](false)
```
instead of:
```
ptr.To(false)
```
I personally think the type annotation is redundant , Would love to hear your opinion :) 
(comment is also relevant for the other changes in the other lines)

Reply:
You mean he can understand it even without explicitly writing Bool?

Reply:
I am not sure if that's true for type like int32 etc

Reply:
for bool you might be right. I didn't think of that

Reply:
for other cases, I personally prefer:
```
ptr.To(int32(42))
```

Reply:
So, when the input is a variable, I think the right way is to forget about declaring the type.
This covers bools, but also all the ints flavours where the argument is a variable (meaning, everywhere but the tests where we have a fixed const that would default to int ).

For the others, I don't have a strong preference and I wasn't able to find if one way is more idiomatic than the other, but I'd keep not using it (so `ptr.To(int32(42))` ) just for consistency.